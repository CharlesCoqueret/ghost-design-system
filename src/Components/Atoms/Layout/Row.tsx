import React, { CSSProperties, PropsWithChildren, ReactElement } from 'react';
import classnames from 'classnames';

export interface IRowProps {
  className?: string;
  style?: CSSProperties;
  width?: CSSProperties['width'];
}

const Row = (props: PropsWithChildren<IRowProps>): ReactElement => {
  const { children, className, style, width } = props;

  return (
    <div className={classnames('gds-layout-row', className)} style={{ width: width, ...style }}>
      {children}
    </div>
  );
};

export default Row;
