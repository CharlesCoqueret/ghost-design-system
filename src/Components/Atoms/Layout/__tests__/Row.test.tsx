import React from 'react';
import { render } from '@testing-library/react';

import Row from '../Row';

describe('Row Component', () => {
  it('Row renders', () => {
    const { container } = render(
      <Row className='CLASSNAME' width='300px' style={{ height: '30%' }}>
        <div id='CHILD1' />
        <div id='CHILD2' />
      </Row>,
    );

    expect(container).toMatchSnapshot();
  });
});
