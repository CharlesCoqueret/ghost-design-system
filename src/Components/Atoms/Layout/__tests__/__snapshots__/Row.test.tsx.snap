// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Component renders properly 1`] = `
<div>
  <div
    class="row CLASSNAME"
    style="height: 30%;"
  >
    <div
      id="CHILD1"
    />
    <div
      id="CHILD2"
    />
  </div>
</div>
`;
